src/install/installer.ts

```diff
@@ -55,34 +55,38 @@ export async function installBrowsersWithProgressBar(packagePath: string, browse
     },
     lockfilePath,
   });
-  const linksDir = path.join(browsersPath, '.links');
+  const linksDir = path.join(registryDirectory, '.links');
 
   try {
     await fsMkdirAsync(linksDir,  { recursive: true });
-    await fsWriteFileAsync(path.join(linksDir, sha1(packagePath)), packagePath);
-    await validateCache(packagePath, browsersPath, linksDir, browserNames);
+    await fsWriteFileAsync(path.join(linksDir, calculateSha1(PACKAGE_PATH)), PACKAGE_PATH);
+    await validateCache(linksDir, browserNames);
   } finally {
     await releaseLock();
   }
 }
 
-async function validateCache(packagePath: string, browsersPath: string, linksDir: string, browserNames?: browserPaths.BrowserName[]) {
+async function validateCache(linksDir: string, browserNames?: BrowserName[]) {
   // 1. Collect used downloads and package descriptors.
   const usedBrowserPaths: Set<string> = new Set();
   for (const fileName of await fsReaddirAsync(linksDir)) {
     const linkPath = path.join(linksDir, fileName);
     let linkTarget = '';
     try {
       linkTarget = (await fsReadFileAsync(linkPath)).toString();
-      const browsersToDownload = await readBrowsersToDownload(linkTarget);
-      for (const browser of browsersToDownload) {
-        const usedBrowserPath = browserPaths.browserDirectory(browsersPath, browser);
-        const browserRevision = parseInt(browser.revision, 10);
+      const linkRegistry = new Registry(linkTarget);
+      for (const browserName of allBrowserNames) {
+        if (!linkRegistry.shouldDownload(browserName))
+          continue;
+        const usedBrowserPath = linkRegistry.browserDirectory(browserName);
+        const browserRevision = parseInt(linkRegistry.revision(browserName), 10);
```

Comment:
Let's move parseInt call to the registry?

Reply:
done